### PR TITLE
#39: add update identity provider and update keycloak identity provid…

### PR DESCRIPTION
**Pull request by sideisra**
…er actions

### CODE REVIEW

File: docsbuild/content/migrations/identityprovider.md [hunk 1]
```diff
@@ -59,6 +59,52 @@ changes:
       trustEmail: true
       storeToken: true
 ```
+## UpdateKeycloakIdentityProvider
+Updates an existing keycloak identity provider.
```
Comment (klg71) [inline docsbuild/content/migrations/identityprovider.md:63]: I would add, and I am quite sure this is the case: `Updates an existing keycloak identity provider leaves all not provided fields untouched.`

File: keycloakapi/src/main/kotlin/de/klg71/keycloakmigration/keycloakapi/KeycloakClient.kt [hunk 2]
```diff
@@ -1,46 +1,6 @@
...
-import de.klg71.keycloakmigration.keycloakapi.model.Role
-import de.klg71.keycloakmigration.keycloakapi.model.RoleListItem
-import de.klg71.keycloakmigration.keycloakapi.model.UpdateFlow
-import de.klg71.keycloakmigration.keycloakapi.model.UpdateFlowExecution
-import de.klg71.keycloakmigration.keycloakapi.model.UpdateGroup
-import de.klg71.keycloakmigration.keycloakapi.model.User
-import de.klg71.keycloakmigration.keycloakapi.model.UserFederation
-import de.klg71.keycloakmigration.keycloakapi.model.UserFederationMapper
-import de.klg71.keycloakmigration.keycloakapi.model.UserGroupListItem
+import de.klg71.keycloakmigration.keycloakapi.model.*
```
Comment (klg71) [inline keycloakapi/src/main/kotlin/de/klg71/keycloakmigration/keycloakapi/KeycloakClient.kt:3]: Star-Imports should be forbidden by the build process but I will check that again

**Review by klg71 — CHANGES_REQUESTED** [review]
Only a few changes requested :)

File: keycloakapi/src/main/kotlin/de/klg71/keycloakmigration/keycloakapi/KeycloakClient.kt (hunk 2 above)
Comment (sideisra) [reply keycloakapi/src/main/kotlin/de/klg71/keycloakmigration/keycloakapi/KeycloakClient.kt:3]: The build complained about a star import in ActionFactory but not in the api.

Comment (sideisra) [reply keycloakapi/src/main/kotlin/de/klg71/keycloakmigration/keycloakapi/KeycloakClient.kt]: done

File: docsbuild/content/migrations/identityprovider.md (hunk 1 above)
Comment (sideisra) [reply docsbuild/content/migrations/identityprovider.md]: done

**Review by klg71 — APPROVED** [review]

**Pull request merged**